components/scream/src/physics/share/physics_saturation_impl.hpp

```diff
@@ -27,22 +27,22 @@ ::check_temperature(const Spack& t_atm, const char* /* func_name */, const Smask
 
   //find out if there are any negative temperatures in the pack
   const auto is_neg_t_atm = (t_atm <= 0) && range_mask;
-  EKAT_KERNEL_REQUIRE_MSG(!(is_neg_t_atm.any()), "Error! Temperature has <= 0 values.\n"); //exit with an error message
+  EKAT_KERNEL_REQUIRE_MSG(!(is_neg_t_atm.any()), caller); //exit with an error message
 
   //find out if there are any NaN temperatures in the pack
   const auto is_nan_t_atm = isnan(t_atm) && range_mask;
-  EKAT_KERNEL_REQUIRE_MSG(!(is_nan_t_atm.any()), "Error! Temperature has NaN values.\n"); //exit with an error message
+  EKAT_KERNEL_REQUIRE_MSG(!(is_nan_t_atm.any()), caller); //exit with an error message
 
 }
 
 template <typename S, typename D>
 KOKKOS_FUNCTION
 typename Functions<S,D>::Spack
-Functions<S,D>::MurphyKoop_svp(const Spack& t_atm, const bool ice, const Smask& range_mask)
+Functions<S,D>::MurphyKoop_svp(const Spack& t_atm, const bool ice, const Smask& range_mask, const char* caller)
 {
 
   //First check if the temperature is legitimate or not
-  check_temperature(t_atm, "MurphyKoop_svp", range_mask);
+  check_temperature(t_atm, ekat::impl::strlen(caller) == 0 ? "MurphyKoop_svp" : caller, range_mask);
```

Comment:
A default to `nullptr` I think is more natural. Then here you wouldn't call `strlen` but instead just check if `caller == `nullptr` (or, idiomatically, simply `caller ? caller : "MurphyKoop_svp"`).